### PR TITLE
ref(ts): Convert worldMapChart to typescript

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
Converts worldmapChart to typescript

### CODE REVIEW

**Review by billyvg — APPROVED** [review]
LGTM

File: src/sentry/static/sentry/app/components/charts/worldMapChart.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import BaseChart from './baseChart';
+import MapSeries from './series/mapSeries';
+import VisualMap from './components/visualMap';
+
+type ChartProps = React.ComponentProps<typeof BaseChart>;
+
+type MapChartSeriesDataUnit = Omit<SeriesDataUnit, 'name' | 'itemStyle'> & {
+  name: string | undefined;
+  // Docs for map itemStyle differ from Series data unit. See https://echarts.apache.org/en/option.html#series-map.data.itemStyle
+  itemStyle: EChartOption.SeriesMap.DataObject['itemStyle'];
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/charts/worldMapChart.tsx:19]:
```suggestion
  // Docs for map itemStyle differ from Series data unit. See https://echarts.apache.org/en/option.html#series-map.data.itemStyle
  itemStyle: EChartOption.SeriesMap.DataObject['itemStyle'];
  name?: string;
```

File: src/sentry/static/sentry/app/components/charts/worldMapChart.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+type MapChartSeries = Omit<Series, 'data'> & {
+  data: MapChartSeriesDataUnit[];
+};
+
+type Props = Omit<ChartProps, 'series'> & {
+  series: MapChartSeries[];
+  seriesOptions?: EChartOption.SeriesMap;
+};
+
+type JSONResult = {[key: string]: any};
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/charts/worldMapChart.tsx:31]:
```suggestion
type JSONResult = Record<string, any>;
```

File: src/sentry/static/sentry/app/components/charts/worldMapChart.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+type State = {
+  countryToCodeMap: JSONResult | null;
+  map: JSONResult | null;
+  codeToCountryMap: JSONResult | null;
+};
+
+class WorldMapChart extends React.Component<Props, State> {
+  static propTypes = {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/charts/worldMapChart.tsx:40]: If I am not mistaken, the `WorldmapChart` is only being used in the `getChartComponent`, which is already n typescript, so it's safe to remove these propTypes 😉

File: src/sentry/static/sentry/app/components/charts/worldMapChart.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    ...BaseChart.propTypes,
+    seriesOptions: PropTypes.object,
+  };
+
+  state = {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/charts/worldMapChart.tsx:45]:
```suggestion
  state: State = {
```

sometimes I need to add the State type here as well. Please does it work fine for you without it?

File: src/sentry/static/sentry/app/components/charts/worldMapChart.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+      codeToCountryMap: Object.fromEntries(
+        Object.entries(countryToCodeMap.default).map(([country, code]) => [code, country])
+      ),
+    });
+  }
+
+  render() {
+    if (this.state.countryToCodeMap === null || this.state.map === null) {
+      return null;
+    }
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/charts/worldMapChart.tsx:72]:
```suggestion
    const {countryToCodeMap, map } = this.state;
    
    if (countryToCodeMap === null || map === null) {
      return null;
    }
```

**Review by priscilawebdev — APPROVED** [review]
I left some suggestions. but it looks good 😉

File: src/sentry/static/sentry/app/components/charts/worldMapChart.tsx (hunk 4 above)
Comment (k-fish) [reply src/sentry/static/sentry/app/components/charts/worldMapChart.tsx:45]: I like to add it regardless, just missed this one.

**Pull request merged**